# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Parent  86f5fb3884458536e2c1b77ac0c0cb3748dcd0ec
Feature: Restrict 'Open Message For ID' to message pane for now. Doesn't work for message in tab.

diff --git a/mail/base/content/nsContextMenu.js b/mail/base/content/nsContextMenu.js
--- a/mail/base/content/nsContextMenu.js
+++ b/mail/base/content/nsContextMenu.js
@@ -418,17 +418,19 @@ class nsContextMenu {
     this.showItem("mailContext-copyemail", this.onMailtoLink);
     this.showItem("mailContext-copylink", this.onLink && !this.onMailtoLink);
     this.showItem("mailContext-copyimage", this.onImage);
 
     this.showItem(
       "mailContext-openMessageForMsgId",
       this.onMailtoLink &&
         !this.inThreadPane &&
-        gFolderDisplay.selectedMessageIsNews
+        this.isNewsgroup &&
+        // This fails badly if message opened in tab.
+        gFolderDisplay._folderPaneVisible
     );
     this.showItem(
       "mailContext-composeemailto",
       this.onMailtoLink && !this.inThreadPane
     );
     this.showItem(
       "mailContext-addemail",
       this.onMailtoLink && !this.inThreadPane
